rust/processor/src/db/common/models/emojicoin_models/constants.rs

```diff
@@ -0,0 +1,53 @@
+use std::env;
+
+pub const SWAP_EVENT: &'static str = concat!(
+    env!("EMOJICOIN_MODULE_ADDRESS"),
+    "::emojicoin_dot_fun::Swap"
+);
+pub const CHAT_EVENT: &'static str = concat!(
+    env!("EMOJICOIN_MODULE_ADDRESS"),
+    "::emojicoin_dot_fun::Chat"
+);
+pub const MARKET_REGISTRATION_EVENT: &'static str = concat!(
+    env!("EMOJICOIN_MODULE_ADDRESS"),
+    "::emojicoin_dot_fun::MarketRegistration"
+);
+pub const PERIODIC_STATE_EVENT: &'static str = concat!(
+    env!("EMOJICOIN_MODULE_ADDRESS"),
+    "::emojicoin_dot_fun::PeriodicState"
+);
+pub const STATE_EVENT: &'static str = concat!(
+    env!("EMOJICOIN_MODULE_ADDRESS"),
+    "::emojicoin_dot_fun::State"
+);
+pub const GLOBAL_STATE_EVENT: &'static str = concat!(
+    env!("EMOJICOIN_MODULE_ADDRESS"),
+    "::emojicoin_dot_fun::GlobalState"
+);
+pub const LIQUIDITY_EVENT: &'static str = concat!(
+    env!("EMOJICOIN_MODULE_ADDRESS"),
+    "::emojicoin_dot_fun::Liquidity"
+);
+pub const MARKET_RESOURCE: &'static str = concat!(
+    env!("EMOJICOIN_MODULE_ADDRESS"),
+    "::emojicoin_dot_fun::Market"
+);
+// When a market is first registered, the market_nonce field is emitted in the resulting events as 1.
+pub const INITIAL_MARKET_NONCE: i64 = 1;
+
+#[cfg(test)]
+mod tests {
+    use crate::utils::util::standardize_address;
+
+    #[test]
+    fn ensure_contract_address_is_standardized() {
+        if standardize_address(env!("EMOJICOIN_MODULE_ADDRESS")) != env!("EMOJICOIN_MODULE_ADDRESS")
+        {
+            panic!(
+                "The non-standardized contract address: {} is invalid because it doesn't match the standardized address: {}",
```

Comment:
Yes basically I just made an assumption that we would standardize the address when passed in as an env var.

I'll be honest, this was like the very first thing I wrote and I didn't fully know where the entrypoint to the processor was lol, so this could/should probably just be a check in `main.rs` or something

Reply:
As I understand it, if we adopt the convention that all addresses will be in "normalized" format (with leading zeroes), then queries to the DB will require users to put leading zeroes correct?

For example in the DB will be `0x0abc`, and PostgREST will need to query `0x0abc`

Just asking because whatever we choose now will need to be consistent

Reply:
Yes I mean I think that's the simplest way to approach it (and probably how it always should have been for the node JSON response 😛). Basically just a simple requirement that addresses are always exactly `0x` + 64 chars, 0-padded on the left if not long enough

Since we don't really utilize the core addresses (`0x0-15`) we're not wasting much space, just making it simpler to store and query